tests/integration/targets/nd_pcv/tasks/file_upload.yml

```diff
@@ -0,0 +1,248 @@
+# Test code for the NAE modules
+# Copyright: (c) 2022, Cindy Zhao (@cizhao) <cizhao@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: Set vars
+  set_fact:
+    nd_info: &nd_info
+      output_level: '{{ nd_output_level | default("info") }}'
+
+- name: Add pre-change analysis from file upload (non JSON format)
+  cisco.nd.nd_pcv:
+    <<: *nd_info
+    ig_name: day2ops-tme
+    site_name: prod-fabric
+    file: targets/nd_pcv/tasks/ansible_dump.json
+    name: ansible_no_json
+    state: present
+  register: pcv_no_json
+
+- name: Verify pcv_no_json
+  assert:
+    that:
+    - pcv_no_json is changed
+    - pcv_no_json.previous == {}
+    - pcv_no_json.current.name == "ansible_no_json"
+    - pcv_no_json.current.assuranceEntityName == "prod-fabric"
+    - pcv_no_json.current.analysisStatus is defined
+    - pcv_no_json.current.analysisScheduleId is defined
+    - pcv_no_json.current.baseEpochId is defined
+    - pcv_no_json.current.fabricUuid is defined
+    - pcv_no_json.current.jobId is defined
+    - pcv_no_json.current.submitterName is defined
+
+# check whether pre-change validation complete
+- name: Query Pre-Change analysis job pcv_no_json
+  cisco.nd.nd_pcv:
+    <<: *nd_info
+    ig_name: day2ops-tme
+    site_name: prod-fabric
+    name: ansible_no_json
+    state: query
+  register: nm_query_ansible_no_json
+  until: nm_query_ansible_no_json.current.analysisStatus == "COMPLETED"
+  retries: 60
+  delay: 10
+
+- name: Pause for 1 minutes to complete pre-change analysis in the previous step
+  pause:
+      minutes: 1
+
+- name: Add pre-change analysis from file upload (JSON format)
+  cisco.nd.nd_pcv:
+    <<: *nd_info
+    ig_name: day2ops-tme
+    site_name: prod-fabric
+    file: targets/nd_pcv/tasks/contract.json
+    name: ansible_json
+    state: present
+  register: pcv_json
+
+- name: Query Pre-Change analysis job pcv_no_json
+  cisco.nd.nd_pcv:
+    <<: *nd_info
+    ig_name: day2ops-tme
+    site_name: prod-fabric
+    name: ansible_json
+    state: query
+  register: nm_query_ansible_json
+  retries: 60
+  delay: 10
+  until: nm_query_ansible_json.current.analysisStatus == "COMPLETED"
+
+- name: Pause for 1 minutes to complete pre-change analysis in the previous step
+  pause:
+      minutes: 1
+
+- name: Add pre-change analysis from file upload with same file but different file name (JSON format)
+  cisco.nd.nd_pcv:
+    <<: *nd_info
+    ig_name: day2ops-tme
+    site_name: prod-fabric
+    file: targets/nd_pcv/tasks/contract_identical.json
+    name: ansible_json
+    state: present
+  register: pcv_json_diff_filename
+  ignore_errors: True
+
+- name: Verify pcv_json pcv_json_diff_filename
+  assert:
+    that:
+    - pcv_json is changed
+    - pcv_json.previous == {}
+    - pcv_json.current.name == "ansible_json"
```

Comment:
This assert is on the previous one not this one (this one would be pcv_json_diff_filename)
Can we add a few more checks?